cache.go

```diff
@@ -139,11 +144,22 @@ func (c *cache) remove(name string) error {
 	return nil
 }
 
+func (c *cache) monitor(tick time.Duration) {
+	ticker := time.NewTicker(tick)
+	for range ticker.C {
+		c.mu.RLock()
```

Comment:
A goroutine is spawned for each `refreshAndLog` so the time the lock is held should be quite small. The lock is only really needed to make sure that no entries are remove while iterating over `c.entries` so its okay to release it before all the refreshes actually finish.